packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPickerUtils/useSvgParser.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react';
+
+import { traverseSvg } from './SvgParserHelper';
+
+interface IConvertComponentProps {
+  selectedColor: string;
+  iconString: string;
+  width?: string;
+  height?: string;
+}
+
+type SvgReturnType = { svgCode: string; width?: string; height?: string };
+
+export const useSvgParser = ({ selectedColor, iconString, width, height }: IConvertComponentProps): SvgReturnType => {
+  const [svgCode, setSvgCode] = useState<string>('');
+
+  useEffect(() => {
+    try {
+      const decodeBase64 = (str: string) => {
+        if (typeof atob === 'function') {
+          return atob(str);
+        } else {
+          return Buffer.from(str, 'base64').toString('binary');
+        }
+      };
+
+      // Extract the SVG element from the base64 encoded string
+      const svgString = decodeBase64(iconString!);
+      const parser = new DOMParser();
+      const svgDocument = parser.parseFromString(svgString, 'image/svg+xml');
+      const svgRoot = svgDocument.documentElement;
+
+      traverseSvg(svgRoot, selectedColor);
+      const modifiedSvg = svgRoot.outerHTML;
+      if (!svgRoot) {
+        throw new Error('Invalid SVG string');
+      }
+      setSvgCode(modifiedSvg);
+    } catch (error) {
+      console.error('Error parsing SVG', error);
+    }
+  }, [selectedColor, iconString]);
+
+  return { svgCode, width, height };
```

Comment:
do we need to return `width, height` again since they are parameters passed in?

Reply:
Those are optional anyways. In case if we need to take this whole component out of scene composer. It would be a better option for anyone to set their own width and height.

Reply:
but the width/height returned is exactly the ones they passed in. they don't need to get the value from the return

Reply:
done